### ISSUE TITLE
fancybox 3.3.0 all bad on iOs

### ISSUE THREAD

**Issue by thomahawk4** [fancyapps/fancybox#1935]
Hello Fancybox Team
I have fancybox 3.3.0 in use on our new website at http://burgerg9.myhostpoint.ch/index.php

It works okay everywhere, except on iOS. Here the popups show only for a small upper part (inline html content), or nothing at all (iframe content). I can not find anyone who describes such problems, so I have no clue why this is or how to solve it. Can you help?

(We also use fancybox for the image gallery, there it works well with the exception that touch outside the image does not close it.)

Greetings
Thom

**Comment by thomahawk4** [comment]
Thank you, that was of course not a clean thing to do. I removed it, but as it shows, it was not the source of the problems I have. Still, and only in iOS, all popups are only shown partial. Some popups do not appear at all (for example the one with this iframe content: http://burgerg9.myhostpoint.ch/baumkarte.html

I think as fancybox works well on Mac, Windows, Android, it must be an iOS issue, right? But how go around it?

regards
Thomas

**Comment by fancyapps** [comment]
Why do you have fancybox files in your iframed page (baumkarte.html) and why you still have `<body class="fancybox-content">`? Please, remove/change that.

**Comment by thomahawk4** [comment]
No, thats not the case anymore. Suppose you have the old page still in your cache. I had changed it to <body class="fancybox-fenster">

**Comment by thomahawk4** [comment]
On this website, overflow is not that necessary because most pages are long, I deleted it. 

AHA!

It seems you just solved my problem!

But overflow it is often necessary because it is ugly when switching between short pages without scrollbars and long pages with scrollbars make the whole template wiggle around. For that reason it is standard on all my websites - and since I have used earlier versions of fancybox on those other websites and did not have such problems there, I did not think about this as a possible reason of the problem – even more so as it was only a problem on iOS.

THANK YOU!

Now the only problem remaining is that in the gallery
http://burgerg9.myhostpoint.ch/galerie.html

the popped up images do not go away when touching outside - but thats on android and iOS the same. It only works on Win/Mac.

**Comment by fancyapps** [comment]
You can use `mobile` option to change action when clicked content/slide/or outside slide, for example:

```JS
mobile: {
    clickContent: function(current, event) {
        return current.type === "image" ? "toggleControls" : false;
    },
    clickSlide: function(current, event) {
        return "close";
    }
}
```

**Comment by thomahawk4** [comment]
Would this be as a separate Javascript?

**Comment by thomahawk4** [comment]
By the way, fancybox has exactly that problem in general and on all plattforms: The whole layout always woggles as soon as a popup opens, because the vertical scrollbar disappears, moves the whole page to the right and then opens the popup and when closing the whole thing again backwards. 

My html: overflow always had prevented this on my other websites, but now as I had to remove the overflow because of iOS, that problem is back and it gives a very unprofessional feeling to the whole website.

**Comment by fancyapps** [comment]
1) About clicking/closing - you see, people on mobile devices are used that tapping outside the image toggles controls (default iOS Photos app and also Facebook, Twitter, etc), therefore fancybox creates similar user experience. If you wish, you can customize, see docs about setting custom options.

2) About "woggling" - the script hides scollbar by applying `overflow:hidden` to `body` tag and then compensates missing space by applying right margin. This works perfectly fine (see homepage and literally every demo) unless you have forced scrollbars to appear. If you wish, you can disable this feature using `hideScrollbar: false` option, but be aware, that double scrollbar can appear for non-image content.

3) About your concern related to page scrollbar. What can I say, wow, this brings back old memories. I remember when I started learning HTML back in the 90's, I was concerned about this too. I felt frustrated, but then I learned that this is nothing and stopped paying attention. If you still want, you can force vertical scrollbars using `body {overflow-y: scroll;}`

**Comment by thomahawk4** [comment]
Your great help is really appreciated. Many thanks! Greetings, Thom

**Comment by thomahawk4** [comment]
Well, I have no idea how to implement it. Like this, the behaviour does not change:

```
<script type="text/javascript">
$(document).ready(function() {
mobile: {
    clickContent: function(current, event) {
        return current.type === "image" ? "toggleControls" : false;
    },
    clickSlide: function(current, event) {
        return "close";
    }
}
});
</script>
```

**Comment by fancyapps** [comment]
That code goes as an argument for `fancybox()` method:

```JS
$("[data-fancybox]").fancybox({
	// Put your options here !!!!
});
```

See http://fancyapps.com/fancybox/3/docs/#options for examples.

**Comment by thomahawk4** [comment]
This did it. Works like a charm now. Thank you!
